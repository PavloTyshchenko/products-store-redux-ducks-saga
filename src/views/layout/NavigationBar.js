import React from 'react';
import { Link } from 'react-router-dom';
import { Navbar, Nav, NavItem } from 'react-bootstrap';
import { LinkContainer } from "react-router-bootstrap";

const NavigationBar = () => {

    return (
        <Navbar bg="primary" variant="dark" className="mb-3">
            <LinkContainer to="/">
                <Navbar.Brand>Navbar</Navbar.Brand>
            </LinkContainer>
            <Nav className="mr-auto">
                <Nav.Link as={Link} to='/'>Home</Nav.Link>
                <Nav.Link as={Link} to='/products/All'>Products</Nav.Link>
                <Nav.Link as={Link} to='/about'>About</Nav.Link>
            </Nav>

        </Navbar>
    );
};

export default NavigationBar;
