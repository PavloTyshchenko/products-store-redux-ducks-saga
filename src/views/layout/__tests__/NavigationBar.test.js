import React from 'react';
import { shallow } from 'enzyme';

import NavigationBar from '../NavigationBar';

describe('NavigationBar Component', () => {

    it('Should render without errors', () => {
        const component = shallow(<NavigationBar />);
        expect(component.length).toBe(1);
    });

    test('Should match snapshot', () => {
        const component = shallow(<NavigationBar />);
        expect(component.debug()).toMatchSnapshot();
    });
});