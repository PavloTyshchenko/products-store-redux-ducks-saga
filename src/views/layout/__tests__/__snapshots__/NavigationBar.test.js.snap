// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar Component Should match snapshot 1`] = `
"<Bootstrap(undefined) bg=\\"primary\\" variant=\\"dark\\" className=\\"mb-3\\">
  <withRouter(LinkContainer) to=\\"/\\">
    <NavbarBrand>
      Navbar
    </NavbarBrand>
  </withRouter(LinkContainer)>
  <Nav className=\\"mr-auto\\" justify={false} fill={false}>
    <NavLink as={[Function: Link]} to=\\"/\\" disabled={false}>
      Home
    </NavLink>
    <NavLink as={[Function: Link]} to=\\"/products/All\\" disabled={false}>
      Products
    </NavLink>
    <NavLink as={[Function: Link]} to=\\"/about\\" disabled={false}>
      About
    </NavLink>
  </Nav>
</Bootstrap(undefined)>"
`;
